src/zenml/integrations/azure/secrets_managers/azure_secrets_manager.py

```diff
@@ -55,6 +65,109 @@ def _ensure_client_connected(cls, vault_name: str) -> None:
+    ) -> None:
+        """Validate the scope and namespace value.
+
+        Args:
+            scope: Scope value.
+            namespace: Optional namespace value.
+        """
+        if namespace:
+            cls.validate_secret_name_or_namespace(namespace, scope)
+
+    @classmethod
+    def validate_secret_name_or_namespace(
+        cls,
+        name: str,
+        scope: SecretsManagerScope,
+    ) -> None:
+        """Validate a secret name or namespace.
+
+        Azure secret names must contain only alphanumeric characters and the
+        character `-`.
+
+        Given that we also save secret names and namespaces as labels, we are
+        also limited by the 256 maximum size limitation that Azure imposes on
+        label values. An arbitrary length of 100 characters is used here for
+        the maximum size for the secret name and namespace.
+
+        Args:
+            name: the secret name or namespace
+            scope: the current scope
+
+        Raises:
+            ValueError: if the secret name or namespace is invalid
+        """
+        if scope == SecretsManagerScope.NONE:
+            # to preserve backwards compatibility, we don't validate the
+            # secret name for unscoped secrets.
+            return
+
+        if not re.fullmatch(r"[0-9a-zA-Z-]+", name):
+            raise ValueError(
+                f"Invalid secret name or namespace '{name}'. Must contain "
+                f"only alphanumeric characters and the character -."
+            )
+
+        if len(name) > 100:
+            raise ValueError(
+                f"Invalid secret name or namespace '{name}'. The length is "
+                f"limited to maximum 100 characters."
+            )
+
+    def validate_secret_name(self, name: str) -> None:
+        """Validate a secret name.
+
+        Args:
+            name: the secret name
+        """
+        self.validate_secret_name_or_namespace(name, self.scope)
+
+    def _create_or_update_secret(self, secret: BaseSecretSchema) -> None:
+        """Creates a new secret or updated an existing one.
+
+        Args:
+            secret: the secret to register or update
+        """
+        if self.scope == SecretsManagerScope.NONE:
+            # legacy, non-scoped secrets
+
+            for key, value in secret.content.items():
+                encoded_key = base64.b64encode(
+                    f"{secret.name}-{key}".encode()
+                ).hex()
+                azure_secret_name = f"zenml-{encoded_key}"
+
+                self.CLIENT.set_secret(azure_secret_name, value)
+                self.CLIENT.update_secret_properties(
+                    azure_secret_name,
+                    tags={
+                        ZENML_GROUP_KEY: secret.name,
+                        ZENML_KEY_NAME: key,
+                        ZENML_SCHEMA_NAME: secret.TYPE,
+                    },
+                )
+
+                logger.debug("Wrote secret: %s", azure_secret_name)
```

Comment:
```suggestion
                logger.debug("Secret `%s` written to the Azure Key Vault.", azure_secret_name)
```

Log output is a bit terse